python/mxnet/gluon/block.py

```diff
@@ -1381,12 +1381,12 @@ def export(self, path, epoch=0, remove_amp_cast=True):
                                       .format(name=name), stacklevel=3)
                     else:
                         arg_dict['aux:%s'%name] = param._reduce()
-        save_fn = _mx_npx.save if is_np_array() else ndarray.save
+        save_fn = _mx_npx.savez if is_np_array() else ndarray.save
```

Comment:
does the parameter class note down whether it was created as np array or legacy ndarray? how are we handling attempts at sharing parameters between the two?

Reply:
npy and npz formats are numpy formats. We can serialize legacy mxnet arrays to the new (more general) numpy semantics, but not necessarily the other way around. What is your question?

Reply:
I'm trying to see if there will ever be the case that the behavior of the quoted statement is wrong. One case I can think of is that parameters are created with the numpy class while the flag of `is_np_array()` returns false for some reason. I imagine this is only possible when there's mixed use such as in parameter sharing, hence the question.

Reply:
Sharing parameters between models expected ndarray and NDArray may not be supported. This would trigger error at model runtime as numpy operator expect ndarray but may get NDArray.

With respect to the `nd.save` function. It calls the same C-API as the prior implementation of `npx.save`. Thus saving ndarrays via `nd.save` works (as ndarray is a subclass of NDArray)